### PR TITLE
chore(content-manager): use documentId over id

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* uses documentId instead of id in the content-manager
* updates associated documentation

### Related issue(s)/PR(s)

* resolves CONTENT-2257

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/services/documents.ts [hunk 1]
```diff
@@ -282,61 +281,59 @@ const documentApi = contentManagerApi.injectEndpoints({
...
-        model: string;
-        /**
-         * You don't pass the ID if the document is a single-type
-         */
-        id?: string;
-        params?: Contracts.CollectionTypes.Unpublish.Request['query'];
-        data: Contracts.CollectionTypes.Unpublish.Request['body'];
-      }
+      Pick<Contracts.CollectionTypes.Unpublish.Params, 'model'> &
+        Partial<Pick<Contracts.CollectionTypes.Unpublish.Params, 'documentId'>> & {
```
Comment (Marc-Roig) [inline packages/core/admin/admin/src/content-manager/services/documents.ts:316]: Don't thing those are clearly exposed by strapi types atm :/ But it feels like Model and DocumentId should be somewhat more generic and imported from somewhere else, but don't know if in this specific case you need this to follow the contract

File: packages/core/admin/admin/src/content-manager/hooks/tests/useDocument.test.tsx [hunk 2]
```diff
@@ -46,7 +47,7 @@ describe('useDocument', () => {
       useDocument({
         collectionType: 'collection-types',
         model: mockData.contentManager.contentType,
-        id: '12345',
+        documentId: '12345',
       })
```
Comment (Marc-Roig) [inline packages/core/admin/admin/src/content-manager/hooks/tests/useDocument.test.tsx:51]: completely unrelated question, would it be possible to infer if the collectionType is collection-types or single-types based on the model?

**Review by Marc-Roig — APPROVED** [review]
🚀

File: packages/core/admin/admin/src/content-manager/services/documents.ts (hunk 1 above)
Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/services/documents.ts:316]: yeah, i think theres some room to be desired here, but at the moment i've just tried to ensure we have one source of truth for the types, i had re-written bits when in reality i should have mutated existing things 🙃

Comment (Marc-Roig) [reply packages/core/admin/admin/src/content-manager/services/documents.ts:316]: makes sense!

File: packages/core/types/src/modules/documents/result/index.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import type { Params } from '..';
 
 type Pagination = { page: number; pageSize: number; pageCount: number; total: number };
 
-export type AnyDocument = { id: Params.Attribute.ID } & { [key: string]: any };
+export type AnyDocument = { documentId: string; id: number } & { [key: string]: any };
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/types/src/modules/documents/result/index.ts:6]: I'm not sure where the notion of Entity.ID being a string ever came from considering it's a number – being the incremental integer on a DB. Would it not be better to say "this is actually what it is"?

File: packages/core/content-manager/shared/contracts/collection-types.ts [hunk 4]
```diff
@@ -324,7 +324,7 @@ export declare namespace BulkPublish {
 export declare namespace BulkUnpublish {
   export interface Request {
     body: {
-      ids: number[];
+      documentIds: number[];
```
Comment (jhoward1994) [inline packages/core/content-manager/shared/contracts/collection-types.ts:327]:
```suggestion
      documentIds: string[];
```

?

Comment (joshuaellis) [reply packages/core/content-manager/shared/contracts/collection-types.ts:327]: yes, they're actually all different which i didn't notice. I've fixed them to be the same! ⭐

**Review by jhoward1994 — APPROVED** [review]

**Pull request merged**